swift/ql/test/query-tests/Security/CWE-022/testPathInjection.swift

```diff
@@ -107,76 +113,88 @@ func test() {
+    fm.getRelationship(UnsafeMutablePointer<FileManager.URLRelationship>.allocate(capacity: 0), of: FileManager.SearchPathDirectory.none, in: FileManager.SearchPathDomainMask(), toItemAt: remoteUrl) // $ hasPathInjection=110
+    let _ = fm.changeCurrentDirectoryPath(remoteString) // $ hasPathInjection=110
+    let _ = fm.unmountVolume(at: remoteUrl, options: [], completionHandler: { _ in }) // $ hasPathInjection=110
+    let _ = fm.NSHFSTypeOfFile(remoteString) // $ hasPathInjection=110
     // Deprecated methods
-    let _ = fm.changeFileAttributes([:], atPath: remoteString) // $ hasPathInjection=104
-    let _ = fm.fileAttributes(atPath: remoteString, traverseLink: false) // $ hasPathInjection=104
-    let _ = fm.fileSystemAttributes(atPath: remoteString) // $ hasPathInjection=104
-    let _ = fm.directoryContents(atPath: remoteString) // $ hasPathInjection=104
-    let _ = fm.createDirectory(atPath: remoteString, attributes: [:]) // $ hasPathInjection=104
-    let _ = fm.createSymbolicLink(atPath: remoteString, pathContent: "") // $ hasPathInjection=104
-    let _ = fm.createSymbolicLink(atPath: "", pathContent: remoteString) // $ hasPathInjection=104
-    let _ = fm.pathContentOfSymbolicLink(atPath: remoteString) // $ hasPathInjection=104
-    let _ = fm.replaceItemAtURL(originalItemURL: remoteNsUrl, withItemAtURL: safeNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=104
-    let _ = fm.replaceItemAtURL(originalItemURL: safeNsUrl, withItemAtURL: remoteNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=104
+    let _ = fm.changeFileAttributes([:], atPath: remoteString) // $ hasPathInjection=110
+    let _ = fm.fileAttributes(atPath: remoteString, traverseLink: false) // $ hasPathInjection=110
+    let _ = fm.fileSystemAttributes(atPath: remoteString) // $ hasPathInjection=110
+    let _ = fm.directoryContents(atPath: remoteString) // $ hasPathInjection=110
+    let _ = fm.createDirectory(atPath: remoteString, attributes: [:]) // $ hasPathInjection=110
+    let _ = fm.createSymbolicLink(atPath: remoteString, pathContent: "") // $ hasPathInjection=110
+    let _ = fm.createSymbolicLink(atPath: "", pathContent: remoteString) // $ hasPathInjection=110
+    let _ = fm.pathContentOfSymbolicLink(atPath: remoteString) // $ hasPathInjection=110
+    let _ = fm.replaceItemAtURL(originalItemURL: remoteNsUrl, withItemAtURL: safeNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=110
+    let _ = fm.replaceItemAtURL(originalItemURL: safeNsUrl, withItemAtURL: remoteNsUrl, backupItemName: nil, options: []) // $ hasPathInjection=110
+}
+
+func testSanitizers() {
+    let remoteString = String(contentsOf: URL(string: "http://example.com/")!)
+
+    let fm = FileManager()
+
+    let filePath = FilePath(stringLiteral: remoteString)
+    if (filePath.lexicallyNormalized().starts(with: FilePath(stringLiteral: "/safe"))) {
```

Comment:
Note: We _do_ have https://github.com/github/codeql/blob/main/swift/ql/lib/codeql/swift/controlflow/BasicBlocks.qll#L272 which you could use to remove this FN. Although, I'm not sure how much effort we want to put into this since we're likely to rewrite this once we have a shared guards library.

Reply:
Just in case this becomes a temporary solution that sticks around more than expected, I followed your advice and used dominance to remove the FN in bf023b0. Thanks for pointing me to `BasicBlocks.qll`.